modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -929,7 +930,11 @@ void cv::gimpl::GStreamingExecutor::setSource(GRunArgs &&ins)
         for (auto &&out_eh : op.nh->outNodes()) {
             out_queues.push_back(reader_queues(*m_island_graph, out_eh));
         }
-        op.isl_exec->handleNewStream();
+
+        if (containsOriginalMeta)
+        {
+            op.isl_exec->handleNewStream();
+        }
```

Comment:
Not sure if this is correct.

Then, calling this method only on `containsOriginalMeta` is also hard to explain. There's a little logic for a reader to understand why do we need to call this at this point.

The logic remains the same - internal states are initialized automatically when Island Executables are recompiled, so no need to call `handleNewStream()` there. It only makes sense to call it when we don't recompile the streams and update meta.

It also needs to be called when the new stream's meta doesn't change and there's no need to reshape, but the stream is still new.

Reply:
Summarizing, you need to call this method here only when your Islands WERE NOT recompiled. `containsOriginalMeta` doesn't capture this in all cases. Please introduce an explicit flag indicating your islands were recompiled by the code above, and run this call only if they weren't.

Reply:
Ideally some comment should occur there to describe this new logic. You can reuse my text above.